src/ucp/core/ucp_ep.inl

```diff
@@ -193,16 +193,15 @@ static inline void ucp_ep_flush_state_reset(ucp_ep_h ep)
 
     ucs_assert(!(ep->flags & (UCP_EP_FLAG_ON_MATCH_CTX |
                               UCP_EP_FLAG_LISTENER)));
-    if (!(ep->flags & UCP_EP_FLAG_FLUSH_STATE_VALID)) {
```

Comment:
but then this check will also happen in release mode?

Reply:
this check was removed from release mode as it doesn't make sense - it is only set and there is no code that resets this flag